SERVER/static/js/chat.js

```diff
@@ -19,33 +20,71 @@
     $(".messages").getNiceScroll(0).resize();
     return $(".messages").getNiceScroll(0).doScrollTop(999999, 999);
   };
+  getInput = function(){
+      /*
+      /gets input the user enters and passes it to sendMessage
+      */
+      var innerText = $.trim($('#text').val());
+      if (innerText !== ""){
+          var packet = {'username':'Adam' , 'message':innerText , 'date':new Date().toUTCString()};
+          postMessage(packet);
+      }
+  };
+
+  postMessage = function(packet) {
+      /*
+      /sends packet to Server
+      */
+      $.ajax({
+          url:'/receive_message',
+          data: JSON.stringify(packet),
+          type: 'POST',
+          contentType: 'application/json',
+          dataType: 'json',
+          success: function(msg){
+            console.log("Good news every one!");
+        }
+    });
+      getMessage();
+  };
 
-  sendMessage = function() {
-    var innerText;
-    innerText = $.trim($("#text").val());
-    var packet = new Object();
-    packet.username = "Adam";
-    packet.message = innerText;
-    packet.date = new Date().toUTCString();
-    var myPacket = JSON.stringify(packet);
-    console.log(myPacket);
-    if (innerText !== "") {
-      $(".messages").append("<li class=\"i\"><div class=\"head\"><span class=\"time\">"+new Date().toUTCString()+"</span><span class=\"name\"></span></div><div class=\"message\">"+innerText+"</div></li>");
+  getMessage = function(){
+      /*
+      /Gets json from Server.
+      */
+      $.ajax({
+          url:'/get_messages',
+          dataType: 'json',
+          type: 'GET',
+          success: function(msg){
+              //if the message/time is in set. ignore.
+              if(!hash.hasOwnProperty(msg.result[msg.result.length-1].username.toString() + msg.result[msg.result.length-1].date.toString() + msg.result[msg.result.length-1].message.toString())){
+                  hash[msg.result[msg.result.length-1].username.toString() + msg.result[msg.result.length-1].date.toString() + msg.result[msg.result.length-1].message.toString()] = true;
+                  showMessage(msg.result[msg.result.length-1].username, msg.result[msg.result.length-1].message, msg.result[msg.result.length-1].date);
+              }
```

Comment:
His very unique solution. It works... that's what matters.

@adamcarlton Add a comment here to explain what you're actually doing.